pkg/controllers/capr/machineprovision/controller.go

```diff
@@ -805,11 +804,13 @@ func getCondition(d data.Object, conditionType string) *summary.Condition {
 	return nil
 }
 
-func constructFilesSecret(driver string, config map[string]interface{}) *corev1.Secret {
+func constructFilesSecret(annotations map[string]string, config map[string]interface{}) *corev1.Secret {
```

Comment:
I'm not crazy about passing all the annotations here but its fine in the context of keeping this PR simple and getting it merged.

Reply:
@apoorvajagtap For both `aliasToPath` and `constructFilesSecret`, passing just alias map looks doable but let me know if you run into issues and I can look into it?

Reply:
I agree, passing annotations could be a bit unnecessary. But, instead of passing the alias map, I preferred passing the annotations["fileToFieldAliases"] string as the input.
This allows the function to encapsulate the parsing logic, keeping the responsibility for interpreting the string within the callee rather than the caller. WDYT?